### ISSUE TITLE
Gauge process is getting killed, when specs folder is executed along with a child spec with unimplemented step

### ISSUE THREAD

**Issue by sguptatw** [getgauge/gauge#457]
**Expected behavior**
Gauge process should not get killed. The behavior should be similar to the one with spec having implemented test.

**Actual behavior**
Getting a null pointer exception and gauge is getting killed
```
[ValidationError] /Users/sswaroop/Downloads/CounterWebApp/g1/specs/example.spec:6: Step implementation not found => 'abcsdef'
# Specification Headings
  ## Abc
java.lang.NullPointerException
	at com.thoughtworks.gauge.processor.ExecuteStepProcessor.process(ExecuteStepProcessor.java:31)
	at com.thoughtworks.gauge.connection.MessageDispatcher.dispatchMessages(MessageDispatcher.java:72)
	at com.thoughtworks.gauge.GaugeRuntime.main(GaugeRuntime.java:37)
java.lang.NullPointerException
        
        Failed Step: abcsdef
        Specification: specs/example.spec:6
        Error Message: Connection closed [127.0.0.1:52369] cause: EOF
        Stacktrace: 
        
        Error Message: write tcp 127.0.0.1:52366->127.0.0.1:52369: use of closed network connection
        Stacktrace: 
        
    Error Message: write tcp 127.0.0.1:52366->127.0.0.1:52369: use of closed network connection
    Stacktrace: 
    
  Error Message: write tcp 127.0.0.1:52366->127.0.0.1:52369: use of closed network connection
  Stacktrace: 
  

Failed to initialize spec datastore. Error: write tcp 127.0.0.1:52366->127.0.0.1:52369: use of closed network connection
# Something

  Error Message: write tcp 127.0.0.1:52366->127.0.0.1:52369: use of closed network connection
  Stacktrace: 
  
Unable to connect to plugin Html Report 2.1.1. proto: required field "SuiteExecutionResult.SuiteResult.SpecResults.ProtoSpec.Items.Scenario.ScenarioItems.Step.StepExecutionResult.PostHookFailure.StackTrace" not set

Specifications:	1 executed	0 passed	1 failed	1 skipped
Scenarios:	1 executed	0 passed	1 failed	2 skipped

Total time taken: 41ms

Process finished with exit code 1
```
**Steps to replicate**
1. Create a spec example.spec in folder spec
```
Specification Headings
=====================
Abc
-----
* abcsdef
```
2. Select the specs folder and the example.spec
3. run the selected option.
4. We get the above exception.

**Version**
```
Gauge version: 0.5.1.nightly-2016-06-16

Plugins
-------
html-report (2.1.1.nightly-2016-06-02)
java (0.4.1.nightly-2016-06-16)
xml-report (0.1.2.nightly-2016-05-27)
```

**Comment by kaustavdm** [comment]
Here `gauge-java` is getting killed because of the `NullPointerException`. The main Gauge process does not get killed. But, it does not get the proper results necessary from the runner. This needs to be fixed on the runners' side.

Can this be reproduced with Ruby and C#?

**Comment by apoorvam** [comment]
Fix should be available in nightly >= 25.07.2016
